src/drivers/imx/sai.c

```diff
@@ -92,8 +97,16 @@ static void sai_start(struct dai *dai, int direction)
 	chan_idx = BIT(0);
 	/* RX3 supports capture on imx8ulp */
 #ifdef CONFIG_IMX8ULP
+	fifo_offset = (dai_fifo(dai, DAI_DIR_PLAYBACK) - dai_base(dai) - REG_SAI_TDR0);
+	if (fifo_offset)
+		fifo_playback = fifo_offset / 4;
```

Comment:
if `fifo_offset` is 0, then `fifo_playback` will be calculated by this formula as 0 too. So, you can remove this `if` and the `fifo_playback` initialisation at the top too. Maybe just do
```
fifo_playback = (dai_fifo(dai, DAI_DIR_PLAYBACK) - dai_base(dai) - REG_SAI_TDR0) >> 2;
```